File: hphp/hack/src/hhbc/hhas_pos.ml
(*
 * Copyright (c) 2017, Facebook, Inc.
 * All rights reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the "hack" directory of this source tree.
 *
 *)

(* Span, emitted as prefix to classes and functions *)
type span = int * int

let pos_to_span pos =
  let (line_begin, line_end, _col_begin, _col_end) =
    Pos.info_pos_extended pos
  in
  (line_begin, line_end)

File: hphp/hack/src/hhbc/hhas_pos.rs
// Copyright (c) Facebook, Inc. and its affiliates.
//
// This source code is licensed under the MIT license found in the
// LICENSE file in the "hack" directory of this source tree.

use oxidized::pos::Pos;

/// Span, emitted as prefix to classes and functions
#[derive(Clone, Debug)]
pub struct Span(pub usize, pub usize);

impl std::convert::From<Pos> for Span {
    fn from(pos: Pos) -> Span {
        let (line_begin, line_end, _, _) = pos.info_pos_extended();
        Span(line_begin, line_end)
    }
}
